src/base/bittorrent/peerinfo.cpp

```diff
@@ -168,8 +168,7 @@ bool PeerInfo::isPlaintextEncrypted() const
 
 PeerAddress PeerInfo::address() const
 {
-    return {QHostAddress(QString::fromStdString(m_nativeInfo.ip.address().to_string()))
-                , m_nativeInfo.ip.port()};
+    return {QHostAddress(m_nativeInfo.ip.data()), m_nativeInfo.ip.port()};
```

Comment:
What is type of m_nativeInfo.ip?

Reply:
ip::tcp::endpoint from boost::asio

Reply:
According to Boost.Asio documentation the data type is dependent on the underlying implementation of the socket layer. I don't think you can use it here.

Reply:
The [real structure](https://www.boost.org/doc/libs/1_71_0/boost/asio/ip/detail/endpoint.hpp) matches `QHostAddress(const sockaddr *sockaddr)`

Reply:
>The real structure matches `QHostAddress(const sockaddr *sockaddr)`

Since when is hoping for undocumented internals of some 3rd party library a good idea? It may differ in different cases.

Reply:
You don't understand, sockaddr is a common struct that is well defined on almost every platform. boost and Qt provides compatibility to the platforms so passing such struct from one to another should be not a problem.